src/ceph-daemon

```diff
@@ -400,15 +402,17 @@ def command_deploy():
     if daemon_type not in ['mon', 'mgr', 'mds', 'osd', 'rgw']:
         raise RuntimeError('daemon type %s not recognized' % daemon_type)
     (config, keyring) = get_config_and_keyring()
-
-    c = get_container(args.fsid, daemon_type, daemon_id)
-    deploy_daemon(args.fsid, daemon_type, daemon_id, c, config, keyring)
+    (uid, gid) = extract_uid_gid()
+    c = get_container(args.fsid, daemon_type, daemon_id, uid, gid)
+    deploy_daemon(args.fsid, daemon_type, daemon_id, c, uid, gid,
+                  config, keyring)
 
 ##################################
 
 def command_run():
     (daemon_type, daemon_id) = args.name.split('.')
-    c = get_container(args.fsid, daemon_type, daemon_id)
+    (uid, gid) = extract_uid_gid()
```

Comment:
This implementation is kinda opinionated already so why don't we also assume ceph-container images are being used? Hardcoding 167 looks acceptable to me.

Reply:
maybe we can assume user=ceph and group=ceph?

Reply:
i suspect downstreams will change the os image.  for now it's extracting it from the image.. we can simplify if its too slow